uavcan/diagnostic/Severity.1.0.uavcan

```diff
@@ -0,0 +1,9 @@
+#
+# Generic message severity representation.
+#
+
+uint2 TRIVIAL  = 0      # No coloring (e.g. white/black)
```

Comment:
Answered above, see Severity.

Reply:
I think what you want to express in a logging situation is not "Severity" but rather a level. And to be useful in logging applications you will at least need to have something that corresponds to the following levels: Error, Warn, Info, Debug. And should probably have a Trace and Fatal as well for completeness.

The problem with this message used for logging is that Warning is the most severe "level" and there is no way to distinguish between Debug and Info (i guess both are trivial).

Reply:
> I think what you want to express in a logging situation is not "Severity" but rather a level.

Correct me if I'm mistaken, but "log level" is a shorthand for "log severity level", so they are essentially the same thing. Perhaps we could use some input from a native speaker here.

> And to be useful in logging applications you will at least need to have something that corresponds to the following levels: Error, Warn, Info, Debug. And should probably have a Trace and Fatal as well for completeness.

The severity levels are defined to closely resemble those of the node status message, although this is admittedly not necessary. Those of the node status message, in turn, were chosen to appease people with the commercial aviation background. Paging @thirtytwobits 

Seeing as the shared bus is an expensive medium (unlike, say, log files), I don't think it will be as useful for messages of extremely low severity, such as Trace or Debug.

Reply:
> Correct me if I'm mistaken, but "log level" is a shorthand for "log severity level", so they are essentially the same thing. Perhaps we could use some input from a native speaker here.

The problem is that when app devs want to map from their internal way of logging (using info, warning, error, etc) they will encounter an unnecessary step of converting between their log levels and the severity levels (and when they're seeking help in comments, all they get are some color names?). Some severities will also have to cater for multiple log levels. Both these things are unfortunate.

> Those of the node status message, in turn, were chosen to appease people with the commercial aviation background

I followed the discussion where we decided to use aviation inspired words for priority levels? When was this discussed in the context of node status? 

> Seeing as the shared bus is an expensive medium (unlike, say, log files), I don't think it will be as useful for messages of extremely low severity, such as Trace or Debug.

I don't think it's anything close to reasonable to have this level turned on by default. But it would be highly useful for bench testing (both debug and trace) and the ability to turn on debug for a misbehaving unit under some controlled tests are also useful.